data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/GetNewBlindedBlock.java

```diff
@@ -26,29 +27,88 @@
+import tech.pegasys.teku.infrastructure.json.types.CoreTypes;
+import tech.pegasys.teku.infrastructure.json.types.DeserializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.json.types.SerializableOneOfTypeDefinitionBuilder;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.json.types.StringValueTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.ParameterMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.datastructures.blocks.BeaconBlock;
 
-public class GetNewBlindedBlock extends GetNewBlock implements Handler {
+@SuppressWarnings("unused")
+public class GetNewBlindedBlock extends MigratingEndpointAdapter {
   private static final String OAPI_ROUTE = "/eth/v1/validator/blinded_blocks/:slot";
   public static final String ROUTE = routeWithBracedParameters(OAPI_ROUTE);
+  private final ValidatorDataProvider provider;
 
-  public GetNewBlindedBlock(final DataProvider dataProvider, final JsonProvider jsonProvider) {
-    super(dataProvider, jsonProvider);
+  private static final ParameterMetadata<UInt64> PARAM_SLOT =
+      new ParameterMetadata<>(
+          SLOT,
+          CoreTypes.UINT64_TYPE.withDescription(
+              "The slot for which the block should be proposed."));
+
+  private static final StringValueTypeDefinition<BLSSignature> SIGNATURE_TYPE =
+      DeserializableTypeDefinition.string(BLSSignature.class)
+          .formatter(BLSSignature::toString)
+          .parser(value -> BLSSignature.fromBytesCompressed(Bytes.fromHexString(value)))
+          .example(
+              "0x1b66ac1fb663c9bc59509846d6ec05345bd908eda73e670af888da41af171505"
+                  + "cc411d61252fb6cb3fa0017b679f8bb2305b26a285fa2737f175668d0dff91cc"
+                  + "1b66ac1fb663c9bc59509846d6ec05345bd908eda73e670af888da41af171505")
+          .description("`BLSSignature Hex` BLS12-381 signature for the current epoch.")
+          .format("byte")
+          .build();
+
+  private static final ParameterMetadata<BLSSignature> PARAM_RANDAO =
+      new ParameterMetadata<>(
+          RestApiConstants.RANDAO_REVEAL,
+          SIGNATURE_TYPE.withDescription(
+              "`BLSSignature Hex` BLS12-381 signature for the current epoch."));
+
+  private static final ParameterMetadata<Bytes32> PARAM_GRAFFITI =
+      new ParameterMetadata<>(
+          GRAFFITI, CoreTypes.BYTES32_TYPE.withDescription("`Bytes32 Hex` Graffiti."));
+  private static final DeserializableTypeDefinition<SpecMilestone> SPEC_VERSION =
+      DeserializableTypeDefinition.enumOf(SpecMilestone.class);
+
+  private final SerializableTypeDefinition<BeaconBlock> BLINDED_BLOCK_RESPONSE_TYPE;
```

Comment:
Do we actually need this? We should be able to just call `request.respondOk` and it will get the type definition for us.

Reply:
ah ok - hadn't done that bit so just assumed i'd need the type available, can remove